coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -487,6 +489,84 @@ protected SqlNode visitGroupBy(ASTNode node, ParseContext ctx) {
     return ctx.grpBy;
   }
 
+
+
+  @Override
+  protected  SqlNode visitGroupingSets(ASTNode node,  ParseContext ctx){
+    // When Hive recognizes a grouping set, it omits the group by node in the constructed AST.
+    // However, Calcite requires the grouping set to be within a group by node when building SqlNode.
```

Comment:
This comment is not very clear. You might depict the structure of the two trees using an example.